From 548178183caaa36c72a56219e124a08ec54edad8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 10 Sep 2021 10:19:26 +0300
Subject: [PATCH] [SW-47990] gaudi2: init if pll

For the I2C slave to be responsive enough while preboot runs from inside
the SRAM, we need to also raise the IF PLLs.
We raise the frequency according to the half nominal table.

Change-Id: I7554536e92ae5c2c6b9e3285745257b4b79a8d79
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 0b065b7..aa71b17 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -15,6 +15,46 @@
 
 static struct hl_pll_cfg gaudi2_pll_cfg[] = {
 	{
+		.name = "DCORE0_IF",
+		.base = mmDCORE0_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.div_factor = { 1, 2, 3, 7 },
+		.div_sel = { HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE1_IF",
+		.base = mmDCORE1_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.div_factor = { 1, 2, 3, 7 },
+		.div_sel = { HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE2_IF",
+		.base = mmDCORE2_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.div_factor = { 1, 2, 3, 7 },
+		.div_sel = { HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE3_IF",
+		.base = mmDCORE3_XBAR_IF_PLL_CTRL_BASE,
+		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.div_factor = { 1, 2, 3, 7 },
+		.div_sel = { HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
 		.name = "DCORE2_SRAM",
 		.base = mmDCORE2_XBAR_BANK_PLL_CTRL_BASE,
 		.nr = 1, .nf = 64, .od = 2, .nb = 4,
